back-end/src/lms/views/students.py

```diff
@@ -176,8 +189,114 @@ def application(self, request: Request, pk=None) -> Response:
+                    (student.university_info.program.faculty.faculty, center_f),
+                    (student.university_info.program.code, center_f),
+                    (student.university_info.program.program, center_f),
+                    (student.university_info.group, center_f),
+                    ('{}, {}, {}'.format(student.recruitment_office.title,
+                                         student.recruitment_office.city,
+                                         student.recruitment_office.district),
+                     center_f),
+                    (student.application_process.
+                     get_medical_examination_display(), center_f),
+                    (student.application_process.get_prof_psy_selection_display(
+                    ), center_f),
+                    ('Да' if student.application_process.preferential_right else
+                     'Нет', center_f),
+                    ('Да'
+                     if student.application_process.characteristic_handed_over
+                     else 'Нет', center_f),
+                    ('Да'
+                     if student.application_process.criminal_record_handed_over
+                     else 'Нет', center_f),
+                    ('Да' if student.application_process.passport_handed_over
+                     else 'Нет', center_f),
+                    ('Да' if student.application_process.
+                     registration_certificate_handed_over else 'Нет', center_f),
+                    ('Да'
+                     if student.application_process.university_card_handed_over
+                     else 'Нет', center_f),
+                    ('Да' if student.application_process.application_handed_over
+                     else 'Нет', center_f),
+                ]
+                # write student info to the sheet
+                for col, (data, cell_format) in enumerate(row_data):
+                    worksheet.write(i, col, data, cell_format)
+
+        workbook.close()
+
+    @extend_schema(parameters=[
+        OpenApiParameter(name='campus',
+                         description='Filter by campus',
+                         required=True,
+                         type=str)
+    ])
+    @action(methods=['get'],
+            url_path='applications/export',
+            detail=False,
+            renderer_classes=[XLSXRenderer])
+    def applications_export(self, request: Request) -> Response:
+        """
+        Send an excel file with info about applicants.
+        Applicants are sorted by campus, specified in request query params.
+        """
+
+        if 'campus' not in request.query_params:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        campus = request.query_params['campus']
+        students = self.queryset.filter(status='AP',
+                                        university_info__campus=campus)
+        self.generate_excel(students)
+        with open('/tmp/export.xlsx', 'rb') as file:
```

Comment:
The same name in two places -- could lead to human errors in the future.

Reply:
That is intended in this case. First, I prepare the variable `students` and then pass it to the method

Reply:
What? I was talking about the name of the file. More than that, concurrent requests will override this file.

Reply:
Will it really? Can you write on a file that is opened for reading? Thought the `xlsxwriter.Workbook(...)` would wait until the file is not being read anymore

Reply:
Why would it wait? It's not library's responsibility to watch for such scenarios.
We should probably generate a random name inside of `self.generate_excel` and return it.

Reply:
Ok, I will do it this way